psi4/src/psi4/libmints/mintshelper.cc

```diff
@@ -3277,6 +3390,30 @@ std::vector<SharedMatrix> MintsHelper::mo_oei_deriv2(const std::string &oei_type
     return mo_grad;
 }
 
+std::vector<SharedMatrix> MintsHelper::mo_overlap_half_deriv1(const std::string &half_der_side, int atom, SharedMatrix C1, SharedMatrix C2) {
+    std::vector<std::string> cartcomp;
+    cartcomp.push_back("X");
+    cartcomp.push_back("Y");
+    cartcomp.push_back("Z");
+
+    std::vector<SharedMatrix> ao_grad;
+    ao_grad = ao_overlap_half_deriv1(half_der_side, atom);
+
+    // Assuming C1 symmetry
+    int nbf1 = ao_grad[0]->rowdim();
+    int nbf2 = ao_grad[0]->coldim();
+
+    std::vector<SharedMatrix> mo_grad;
+    for (int p = 0; p < 3; p++) {
+        std::stringstream sstream;
+        sstream << "mo_overlap_half_deriv1_" << atom << cartcomp[p];
+        SharedMatrix temp(new Matrix(sstream.str(), nbf1, nbf2));
```

Comment:
Ditto, the preferred idiom is `auto temp = std::make_shared<Matrix>(sstream.str(), nbf1, nbf2));`